src/grid/becke.py

```diff
@@ -27,6 +28,25 @@
 class BeckeWeights:
     """Becke weights functions holder class."""
 
+    def __init__(self, radii, order=3):
```

Comment:
In line with the other discussion, it would be good to make radii optional. If not provided, a series of fallbacks can be used to construct them:
1) Take them from the atomic grids, if they have a radius defined. (There are scenarios when this is not the case.)
2) Use some default radius for each chemical element. This implies that atomic numbers are passed to the `generate_weights` method. This argument would also be needed for the Hirshfeld partitioning.

Reply:
Having a dictionary for the radii would make the most sense. You want to avoid any molecule-specific information to go into the constructor, such that the same definition of the weights can be used to different molecules.